stock_storage_type_putaway_strategy/models/stock_location.py

```diff
@@ -0,0 +1,200 @@
+            lambda slst: package_storage_type in slst.package_storage_type_ids
+        )
+        allowed_location_storage_types = self.filter_restrictions(
+            matching_location_storage_types, quants, products
+        )
+        return (
+            not self.allowed_location_storage_type_ids or allowed_location_storage_types
+        )
+
+    def filter_restrictions(self, matching_location_storage_types, quants, products):
+        allowed_location_storage_types = self.env["stock.location.storage.type"]
+        for location_storage_type in matching_location_storage_types:
+            if self._filter_properties(
+                location_storage_type, quants, products
+            ) and self._filter_capacity(location_storage_type, quants):
+                allowed_location_storage_types |= location_storage_type
+        return allowed_location_storage_types
+
+    def _filter_properties(self, location_storage_type, quants, products):
+        if location_storage_type.only_empty:
+            if not self._existing_quants() and not self._existing_planned_moves():
+                return location_storage_type
+        elif location_storage_type.do_not_mix_products:
+            if location_storage_type.do_not_mix_lots:
+                lot = quants.mapped('lot_id')
+                if len(lot) > 1:
+                    return False
+                if not self._existing_quants(
+                    products=products, lot=lot
+                ) and not self._existing_planned_moves(products=products, lot=lot):
+                    return location_storage_type
+            else:
+                if not self._existing_quants(
+                    products=products
+                ) and not self._existing_planned_moves(products=products):
+                    return location_storage_type
+        else:
+            return location_storage_type
+
+    def _filter_capacity(self, location_storage_type, quants):
+        if self._max_height_allowed(
+            location_storage_type, quants
+        ) and self._max_weight_allowed(location_storage_type, quants):
+            return location_storage_type
+        else:
+            return self.env["stock.location.storage.type"]
+
+    def _max_height_allowed(self, location_storage_type, quants):
+        height = quants.package_id.height
+        max_height = location_storage_type.max_height
+        return not (max_height and height and height > max_height)
+
+    def _max_weight_allowed(self, location_storage_type, quants):
+        pack_weight = quants.package_id.pack_weight
+        max_weight = location_storage_type.max_weight
+        return not (max_weight and pack_weight and pack_weight > max_weight)
+
+    def _existing_quants(self, products=None, lot=None):
+
+        base_domain = [("location_id", "=", self.id)]
+        domain = self._prepare_existing_domain(base_domain, products=products, lot=lot)
+        return self.env["stock.quant"].search(domain, limit=1)
```

Comment:
```suggestion
        return bool(self.env["stock.quant"].search(domain, limit=1))
```

but it's not very relevant is just to return on boolean like we expect with method `_existing_*` but don't change if you are not agree :wink:

Reply:
When I read `_existing_*`, I expect to get the existing records. IMO your suggestion would be valid if it red `_is_existing_*` but it's not case.